### PR TITLE
Update developer tools page

### PR DESCRIPTION
**Pull request by YayunHuang**
Ref: MUP-123

browser:

https://github.com/user-attachments/assets/682ea88d-2517-45bf-9f02-3ff7e66f2754

iphone xe:

https://github.com/user-attachments/assets/25e84ff7-44f3-483a-9320-8def715062bc

ipad air:

https://github.com/user-attachments/assets/1a0b5ec3-3882-4797-b94a-ab9487d65331

### CODE REVIEW

**Comment by YayunHuang** [comment]
@pkong-ds I'm not sure why the `transform` in `@keyframes`  doesn't work, but I tested on [CodePen](https://codepen.io/yayun146/pen/eYQxPVK) that it works, do you know the possible reason?

**Comment by pkong-ds** [comment]
@YayunHuang what did not work for you? I tested on firefox and chrome both work

https://github.com/user-attachments/assets/9893fc62-bcce-4213-a1a1-a9d182b9a334

Apple M2 14.2.1
Firefox 129.0 (64-bit)
Chrome 127.0.6533.90

File: src/pages/developerTools.astro [hunk 1]
```diff
@@ -8,38 +9,60 @@ import BaseLayout from "../layouts/BaseLayout.astro";
...
+              <p>
+                Empower Your Team with AI: Eliminate Manual Work, Embrace
+                Seamless Automation!
+              </p>
+              <div class="learn-more">
+                Learn more
+                <img src="Images/arrow-forward.svg" />
+              </div>
+            </div>
           </a>
```
Comment (pkong-ds) [inline src/pages/developerTools.astro]: Maybe we can extract this as a separate component `<DeveloperToolCard />` ? 🙏

**Comment by YayunHuang** [comment]
@pkong-ds I expect the card should transform from `translateY(10%)` but it seems always from 100%

File: src/styles/developer-tools.css [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  }
+}
+
+.developer-tools-card:hover {
+  cursor: pointer;
+  box-shadow: 0px 4px 24px rgba(0, 0, 0, 0.16);
+}
+
+.developer-tools-card.visible {
+  animation: slideInFromBottom 0.3s forwards;
```
Comment (pkong-ds) [inline src/styles/developer-tools.css]: Should we use animation-easing here? maybe `ease-in-out` would do?

ref cubic-bezier-easing-function

File: src/styles/developer-tools.css [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+@-webkit-keyframes slideInFromBottom {
+  0% {
+    transform: translateY(10%);
+    opacity: 0;
+  }
+  100% {
+    transform: translateY(0);
+    opacity: 1;
+  }
+}
```
Comment (pkong-ds) [inline src/styles/developer-tools.css]: Why do we need different keyframes for `moz` and `webkit` here? What problem are we trying to solve?

Can add comment to explain if have reasons 🙏

**Comment by pkong-ds** [comment]
@YayunHuang Maybe check check this commit https://github.com/pkong-ds/mockuphone.com/commit/6a1097ce38e984dc3973977c5154671108919170 and see if it helps~

Major changes

1. Keep only 1 keyframe
2. Do not rely on `.visible`, apply the animation on page load 
    - The intersection observing logic is not necessary, since our page height currently show those developer tool cards directly
    - That way, we can use pure css

> why the transform in @keyframes doesn't work

As for this question, my wildguess is:
- your original css `translateY` _always_ calculate based on `window` instead of `element` itself
- in codepen, it worked because the window height is similar to the card, which mislead you into thinking it "worked"

Open to any other suggestion!

**Comment by YayunHuang** [comment]
@pkong-ds thanks! let me try try,
the`.visible` is for mobile mode, I think the card should only scroll in when the user scrolls to that section, so still need the observing logic?

**Comment by pkong-ds** [comment]
> the `.visible` is for mobile mode

I see! If so, maybe you can put those css inside media queries `@media (max-width: 768px) {`? (btw why 768px? 😆 )

**Comment by YayunHuang** [comment]
@pkong-ds after some testing, the animation seems not reading the value of `@keyframes` at all, even if I remove all `@keyframes` animations are the same :smiling_face_with_tear:

**Comment by pkong-ds** [comment]
🫨 Did you rebuild astro? Maybe the inline stylesheets is somehow cached by cdn?

**Comment by YayunHuang** [comment]
yes I tried this before and other changes besides the animation were applied directly

**Comment by YayunHuang** [comment]
oh I solve the problem, seems we cannot use `slideInFromBottom` as keyframes name :melting_face:

**Comment by YayunHuang** [comment]
@pkong-ds I add some logic to make the animation look smoother in mobile mode, please review, thanks

File: public/scripts/developerTools.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    entries.forEach((entry) => {
+      if (entry.isIntersecting) {
+        entry.target.classList.add("animate");
+      }
+    });
+  }
+
+  const observer = new IntersectionObserver(handleIntersection);
+
+  /* make the animation in mobile mode look smoother */
```
Comment (pkong-ds) [inline public/scripts/developerTools.js]: Why does this make the animation smoother? What problem did it solve?

Comment (YayunHuang) [reply public/scripts/developerTools.js]: @pkong-ds updated comment

File: src/styles/developer-tools.css [hunk 5]
```diff
@@ -0,0 +1,11 @@
+.developer-tools-page {
```
Comment (pkong-ds) [inline src/styles/developer-tools.css]: Sor, just thought of this.

Should we move `developer-tools.css` to `src/components/DeveloperToolCard`?

Maybe we should also refactor existing codebase to be component-isolated too. But that's another issue

**Review by pkong-ds — CHANGES_REQUESTED** [review]

File: src/styles/developer-tools.css (hunk 5 above)
Comment (YayunHuang) [reply src/styles/developer-tools.css]: @pkong-ds but this is for the page, not the `DeveloperToolCard` component, that's why I didn't move it to `src/components/DeveloperToolCard`, how about renamed the folder to `src/components/DeveloperTools` and put  `developer-tools.css` in?

Comment (pkong-ds) [reply src/styles/developer-tools.css]: @YayunHuang I am thinking below. Any thoughts?

```
src/
   pages/
       DeveloperToolPage/
           DeveloperToolPage.astro // or index.astro
           developer-tool-page.css // or index.css
           developer-tool-page.js // or index.js
           DeveloperToolCard.astro
       // later move all pages here. one page one folder
```

Comment (YayunHuang) [reply src/styles/developer-tools.css]: sound nice!! let me update it and create an issue for other pages~~

Comment (YayunHuang) [reply src/styles/developer-tools.css]: @pkong-ds rebased to main and update, but the script must put in `public/scripts`, so I only move `.astro` and `.css` file~

Comment (pkong-ds) [reply src/styles/developer-tools.css]: ah, scripts import 😞 ref https://linear.app/oursky/issue/MUP-107/[code-quality]-refactoring

Will merge this first then 🙏

**Review by pkong-ds — APPROVED** [review]

**Pull request merged**